src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -751,9 +751,11 @@ private static HttpHeaders createPseudoHeaders(HttpRequest request) {
         hdrs.setHeader(":method", method);
         URI uri = request.uri();
         hdrs.setHeader(":scheme", uri.getScheme());
-        // TODO: userinfo deprecated. Needs to be removed
-        hdrs.setHeader(":authority", uri.getAuthority());
-        // TODO: ensure header names beginning with : not in user headers
+        if (uri.getHost() != null && uri.getPort() != -1) {
```

Comment:
I would suggest creating local variables for `host` and `port` in order to avoid calling getHost/getPort multiple times. It will also make possible to `assert host != null` since that's what we expect (it should not reach here if getHost() returns null)